lib/blockscore/actions/retrieve.rb

```diff
@@ -12,7 +12,7 @@ module Actions
     module Retrieve
       module ClassMethods
         def retrieve(id, options = {})
-          fail ArgumentError if id.empty?
+          fail ArgumentError,'ID must be supplied' if id.nil? || id.empty?
```

Comment:
I like the whitelisting approach and was savvy to the \A and \z already. However I need to check with @backus as whitelisting - while useful IMO - requires me to be privy to their id scheme. Example Id is "54d956c06230620003760500". The upside is that it is prechecked (perhaps also for length?) and rejected. Downside - at least one I can think of - is if the ID regex changes it also means an API code change.